MediaBrowser.Providers/Manager/ProviderManager.cs

```diff
@@ -274,12 +276,20 @@ public async Task<IEnumerable<RemoteImageInfo>> GetAvailableRemoteImages(BaseIte
                     result = result.Where(i => i.Type == type.Value);
                 }
 
-                if (preferredLanguages.Count > 0)
+                if (!includeAllLanguages && preferredLanguages.Count > 0)
                 {
+                    // Filter out languages that do not match the preferred languages.
+                    //
+                    // TODO: should exception case of "en" (English) eventually be removed?
                     result = result.Where(i => string.IsNullOrEmpty(i.Language) ||
                                                preferredLanguages.Contains(i.Language, StringComparer.OrdinalIgnoreCase) ||
                                                string.Equals(i.Language, "en", StringComparison.OrdinalIgnoreCase));
                 }
+                else if (preferredLanguages.Count > 0)
+                {
+                    // Order the images by preferred languages first.
+                    result = result.OrderBy(i => !string.IsNullOrEmpty(i.Language) && preferredLanguages.Contains(i.Language, StringComparer.OrdinalIgnoreCase) ? 0 : 1);
```

Comment:
I'm not sure. Most providers should already be using https://github.com/jellyfin/jellyfin/blob/e9524f89d63894ea9af62cd1b61ddd89cb8b9e82/MediaBrowser.Model/Extensions/EnumerableExtensions.cs#L19

Reply:
Ah yep, that seems to do what we want here anyways. Pretty easy change made.